code/modules/power/stirling.dm

```diff
@@ -0,0 +1,218 @@
+#define HEAT_TRANSFER 5000
+#define MAX_FREQUENCY 60
+#define MIN_DELTA_T 5
+
+/obj/machinery/atmospherics/binary/stirling
+	name = "stirling engine"
+	desc = "A mechanical stirling generator. It generates power dependent on the temperature differential between input gas and a radiative heatsink"
+	icon = 'icons/obj/power.dmi'
+	icon_state = "stirling"
+	density = 1
+	anchored = 1
+	layer = STRUCTURE_LAYER
+	use_power = POWER_USE_OFF
+	base_type = /obj/machinery/atmospherics/binary/stirling
+	construct_state = /decl/machine_construction/default/panel_closed
+	uncreated_component_parts = null
+	stat_immune = NOSCREEN | NOINPUT | NOPOWER // Mechanical machine and display.
```

Comment:
It's not correct to use `NOPOWER` immune together with `use_power` stuff, I believe: `NOPOWER` immunity will circumvent certain logic. My guess is that if you set your machine to have `0` as power draw (likely default) and take off `NOPOWER` nothing will change (you may need to also toggle `interact_offline` or similar, but probably not).

Reply:
I replaced references to power_use with a seperate ``active`` var.